File: telegram-service/src/main/java/de/nak/home_assistant/models/service/devices/Thermostat.java
package de.nak.home_assistant.models.service.devices;

import com.pengrad.telegrambot.model.request.InlineKeyboardButton;
import de.nak.home_assistant.models.telegram.CallbackActions;
import de.nak.home_assistant.models.telegram.CallbackData;

public class Thermostat {

    public static String toString(String name, Boolean on, float temperatur, float istTemperatur, float sollTemperatur) {
        String nameString = "🔥 Heizung: %s\n";
        String statusString = (on ? "⚪️" : "⚫️") + " Status: %s\n";
        String tempString = "🌡 Temperatur: %s°C\n";
        String temp2String = "📊 Ist: %s°C, Soll: %s°C";

        return String.format(nameString + statusString + tempString + temp2String, name, on ? "on" : "off", temperatur, istTemperatur, sollTemperatur);
    }

    public static InlineKeyboardButton toInlineButton(String id, Boolean on, float istTemperatur, int moduleId, String name) {
        CallbackData cbData = new CallbackData()
                .setAction(CallbackActions.SELECT_DEVICE)
                .setId(moduleId, id);

        return new InlineKeyboardButton((on ? "⚪️ " : "⚫️ ") + name + " (" + istTemperatur + "°C)")
                .callbackData(cbData.toJson());
    }
}

File: packages/telegram-service/src/devices/thermostat.ts
import { InlineKeyboardButton } from "node-telegram-bot-api";
import { Device } from "@microhome/types";
import { i18n } from "../i18n";
import { createCallbackData } from "../services/callback-data";

export const thermostatToString = ({
  name,
  on,
  temperatur,
  istTemperatur,
  sollTemperatur,
}: Device): string => {
  const translations = i18n();

  const nameString = `🔥 ${translations.devices.thermostat}: ${name}\n`;
  const statusString =
    (on ? "⚪️" : "⚫️") +
    ` ${translations.devices.status}: ${
      on ? translations.devices.on : translations.devices.off
    }\n`;
  const tempString = `🌡 ${translations.devices.temperature}: ${temperatur}°C\n`;
  const temp2String = `📊 ${translations.devices.is}: ${istTemperatur}°C, ${translations.devices.should}: ${sollTemperatur}°C`;

  return nameString + statusString + tempString + temp2String;
};

export const thermostatToInlineButton = async (
  { id, on, istTemperatur, name }: Device,
  moduleId: number
): Promise<InlineKeyboardButton> => {
  const cbData = await createCallbackData({
    action: "SELECT_DEVICE",
    moduleId,
    deviceId: id,
  });

  return {
    text: `${on ? "⚪️" : "⚫️"} ${name} (${istTemperatur}°C)`,
    callback_data: JSON.stringify(cbData.id),
  };
};
